internal/controller/postgrescluster/pgadmin_test.go

```diff
@@ -31,6 +32,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
```

Comment:
:thinking: Curious about the use of `fake` here vs. `env` test, especially considering we've favored `envtest` per the recommendation for `envtest` in the [go docs for `fake`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client/fake):

> When in doubt, it's almost always better not to use this package and instead use envtest.Environment with a real client and API server.

I suppose this might be a scenario where there is no doubt (and I do see one other existing test that uses `fake`) since the Kube API interaction is limited to a single `Get()`, and we're not actually creating anything and not exec'ing into an actual Pod?

I suppose I had been thinking _any_ Kube API interaction would be good reason to hit a real API server via `envtest`, but if there are scenarios where we feel like it might make sense to reach for `fake` instead (e.g. assuming here this was done to keep the test simple for what is a simple/standard API interaction), would definitely be worth discussing a bit among the team.

Reply:
> I suppose this might be a scenario where there is no doubt (and I do see one other existing test that uses `fake`) since the Kube API interaction is limited to a single `Get()`, and we're not actually creating anything and not exec'ing into an actual Pod?

Right. I want the Pod in particular states, but perhaps `fake` is an indication that this method is doing too much. Its responsibility right now is to find a Pod, pass an exec to the other package, and hash/cache the result. 

We could isolate the API call by making it from another method and passing the Pod into this one. 🤔 How would we test that method then? A basic exists + not-exists using `envtest` makes sense, but it'd be nice to cut out the other stuff about PodExec and the result. 🤔 Maybe it's enough to no-op PodExec.

> I suppose I had been thinking _any_ Kube API interaction would be good reason to hit a real API server via `envtest`, but if there are scenarios where we feel like it might make sense to reach for `fake` instead (e.g. assuming here this was done to keep the test simple for what is a simple/standard API interaction), would definitely be worth discussing a bit among the team.

This one is looking for a Pod and trying to PodExec. 🤔 I'm pretty sure to get either of those from the API requires a real cluster. Perhaps coverage of these calls is best reserved for an E2E?

💭 Correct me if I'm wrong, but I think we're comfortable with mocking out PodExec. Perhaps something about mocking out Pod API objects?

💭 If we did all our observations up front, this one `Get()` would not be happening here.

Reply:
We should be able to create/mock a Pod using just `envtest`, e.g. as was done here:

https://github.com/CrunchyData/postgres-operator/blob/a2c8ad911b7add6134dcbd1f412e35eb2d220de4/internal/controller/postgrescluster/pgbackrest_test.go#L2179-L2198

However, no denying that is more cumbersome than using `fake`.  And while I agree moving the API call could help in one sense (e.g. no need to worry about), like you said, that just moves the problem elsewhere.

So overall I have no issue with using `fake`; just seems like something worth discussing a bit with the team so we have an idea of when/where its use might make sense (and def no need to change anything for this PR).

And great point about this once again being a place where up-front observation could help us architecturally (e.g. with testing) - definitely agree there.   Maybe we start flagging these points in the code (also going to look at getting a story creating to track our thoughts)?